### ISSUE TITLE
Accessibility Issue: "Unlabel" for Year Navigation Slider in "Browse Artifacts" - TalkBack Enabled

### ISSUE THREAD

**Issue by rzambroni** [gskinnerTeam/flutter-wonderous-app#187]
While testing the Wonderous app on a physical Pixel 8 Pro device running Android 15, we encountered an accessibility issue when using the TalkBack feature. In the "Browse Artifacts" section, there is a horizontal slider at the bottom of the screen that allows users to navigate between different years to view corresponding artifacts. However, TalkBack is not recognizing this slider properly; it announces it as "Unlabel" instead of providing meaningful context to the user likely due to the Semantics widget not being properly configured around the slider. 

--------

**Steps to Reproduce:**

1. Clone the Wonderous app [GitHub - gskinnerTeam/flutter-wonderous-app: A showcase app for the Flutter SDK. Wonderous will educate and entertain as you uncover information about some of the most famous structures in the world.](https://github.com/gskinnerTeam/flutter-wonderous-app.git) 

2. Run it on a device (We tested with a Pixel 8 Pro)

3. Enable TalkBack for accessibility.

4. Navigate to the "Browse Artifacts" section.

5. Attempt to interact with the year navigation slider at the bottom of the screen.

----------

Expected Result: TalkBack should announce a descriptive label for the slider, helping the user understand its function and how to interact with it.

Actual Result: TalkBack announces "Unlabel" for the slider, providing no useful information to the user.

Video Evidence:

https://github.com/user-attachments/assets/e985ad23-1ef3-4dbf-8dbb-aacbabefeb7a

**This happens on Flutter Master and Flutter Stable**

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel master, 3.25.0-1.0.pre.219, on macOS 14.6.1 23G93 darwin-arm64, locale en-US)
    • Flutter version 3.25.0-1.0.pre.219 on channel master at /Users/josefinadiz/flutter/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 6fe09872b1 (4 days ago), 2024-08-30 19:53:11 -0400
    • Engine revision 2d56e44888
    • Dart version 3.6.0 (build 3.6.0-198.0.dev)
    • DevTools version 2.39.0

[✓] Android toolchain - develop for Android devices (Android SDK version 35.0.0)
    • Android SDK at /Users/josefinadiz/Library/Android/sdk
    • Platform android-35, build-tools 35.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.11+0-17.0.11b1207.24-11852314)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 16.1)
    • Xcode at /Applications/Xcode-beta.app/Contents/Developer
    • Build 16B5001e
    • CocoaPods version 1.15.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2024.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.11+0-17.0.11b1207.24-11852314)

[✓] VS Code (version 1.92.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.96.0

[✓] Connected device (4 available)            
    • Pixel 8 Pro (mobile)            • 3A060DLJG002D9        • android-arm64  • Android 15 (API 35)
    • macOS (desktop)                 • macos                 • darwin-arm64   • macOS 14.6.1 23G93
      darwin-arm64
    • Mac Designed for iPad (desktop) • mac-designed-for-ipad • darwin         • macOS 14.6.1 23G93
      darwin-arm64
    • Chrome (web)                    • chrome                • web-javascript • Google Chrome
      128.0.6613.114

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

</details>

**Comment by esDotDev** [comment]
Thanks for the report, we'll take a look.
